### PR TITLE
chore: added code of conduct

### LINKED ISSUE #2129: Set up Community Standards
We should complete [the community standards](https://github.com/FuelLabs/fuels-ts/community) for the repo.

### PR DESCRIPTION
**Pull request by petertonysmith94**
- Closes #2129 

# Summary

- Using [`fuel-ui`](https://github.com/FuelLabs/fuel-ui/blob/master/CODE_OF_CONDUCT.md) for reference, added a code of conduct, using the latest (v2.1) Contributor Covenant Code of Conduct.

# Checklist

- [ ] I **_added_** — `tests` to prove my changes
- [ ] I **_updated_** — all the necessary `docs`
- [X] I **_reviewed_** — the entire PR myself, using the GitHub UI
- [ ] I **_described_** — all breaking changes and the Migration Guide

### CODE REVIEW

**Review by maschad — CHANGES_REQUESTED** [review]
This is an [empty file,](https://github.com/FuelLabs/fuels-ts/pull/2737/files#diff-ffdbe3a1e7ee93cacfc080b6c635ccf3a8f6b0f00f2fb884f78c6b5f9dac8fd2) is this intentional?

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official email address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+[our Discord](https://discord.com/invite/xfpK4Pe).
```
Comment (Torres-ssf) [inline CODE_OF_CONDUCT.md:63]:
```suggestion
reported to the community leaders responsible for enforcement on
[our Discord](https://discord.com/invite/xfpK4Pe).
```

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+- Publishing others' private information, such as a physical or email address,
+  without their explicit permission
+- Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+## Enforcement Responsibilities
+
+Community leaders are responsible for clarifying and enforcing our standards of
+acceptable behavior and will take appropriate and fair corrective action in
+response to any behavior that they deem inappropriate, threatening, offensive,
```
Comment (Torres-ssf) [inline CODE_OF_CONDUCT.md:43]:
```suggestion
response to any behavior deemed inappropriate, threatening, offensive,
```

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
+# Contributor Covenant Code of Conduct
+
+## Our Pledge
+
+We as members, contributors, and leaders pledge to make participation in our
```
Comment (Torres-ssf) [inline CODE_OF_CONDUCT.md:5]:
```suggestion
We, as members, contributors, and leaders, pledge to make participation in our
```

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
👏

**Review by danielbate — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Review by maschad — APPROVED** [review]
🤝

**Pull request merged**